pyxem/tests/signals/test_diffraction_vectors.py

```diff
@@ -123,27 +124,145 @@ def test_get_cartesian_coordinates(diffraction_vectors_map):
 
 
 class TestInitVectors:
-    def test_from_peaks(self):
-        vectors = np.random.randint(0, 100, (2, 2, 10, 2))
+    @pytest.fixture()
```

Comment:
@sivborg I think that is a good point and something I've never really looked into!

1. From a readability (and reliability) standpoint I really like the fact that the fixture is run every time.  This helps solve some weird bugs that might be possible if the same variable is used multiple times in some test.  With multithreaded testing I'm not really sure what happens then but I imagine there could be some really weird bugs. 
2. pytest supports [caching](https://docs.pytest.org/en/6.2.x/fixture.html#fixtures-can-be-requested-more-than-once-per-test-return-values-are-cached) of fixture values! This is super cool and means that the function is only run once for every set of inputs and then the result is cached for when it is needed again. 

This might be something good to add to the dev guide.

Reply:
1. Yes, that does make things more understandable and avoids unintended state changes! The use of a scope might not be necessary in most circumstances, as it may only make a very small time difference for smaller data structures. Better avoid premature optimization! But makes me wonder what we could do if we wanted to remove unnecessary fixture runs, while keeping them where they're needed. Perhaps make two versions of the fixture, a "constant" version that is only run once and cannot be modified, and another regular fixture which can be modified. Then the implementer can choose what would fit the test. But maybe this discussion is better saved for tests that are taking longer time.
2. I didn't know that! Very tidy indeed! Though it seems to be limited to only caching for the same test, meaning it will still be run at least once per test. Maybe there's some functionality to cache between tests?

Reply:
> * Yes, that does make things more understandable and avoids unintended state changes! The use of a scope might not be necessary in most circumstances, as it may only make a very small time difference for smaller data structures. Better avoid premature optimization! But makes me wonder what we could do if we wanted to remove unnecessary fixture runs, while keeping them where they're needed. Perhaps make two versions of the fixture, a "constant" version that is only run once and cannot be modified, and another regular fixture which can be modified. Then the implementer can choose what would fit the test. But maybe this discussion is better saved for tests that are taking longer time.

Just for reference here is the [scope](https://docs.pytest.org/en/6.2.x/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session) parameter for fixtures.  I don't think I really appreciated it before.


> * I didn't know that! Very tidy indeed! Though it seems to be limited to only caching for the same test, meaning it will still be run at least once per test. Maybe there's some functionality to cache between tests?

I think that the scope parameter does exactly that but I might need to test it a bit